### DISCUSSION TITLE
Unused elements

### DISCUSSION THREAD

**Discussion by mporracindie** [apache/superset#22856] [Ideas]
**Is your feature request related to a problem? Please describe.**
Finding unused charts or datasets

**Describe the solution you'd like**
A filter that checks if a chart is being used in a dashboard or if a dataset has any charts created with

**Comment by rusackas** [comment]
Converted this to an "Ideas" thread since we're trying to keep Issues to bug reports. 

Anyway... a couple of ideas to kick-start some conversation.

@john-bodley might have some tips (or maybe even scripts) here, since he's undertaken similar efforts.

@kasiazjc has done a lot of work in surfacing chart/dashboard cross-relationships and might now a clever place to add some "this is not used anywhere" element into the UI.

Also pinging @yousoph and @cccs-RyanK here in case this has useful tie-in with the frontend tagging feature (e.g. auto-tagging things like "orphan" items or "stale" things nobody's viewed in a year)

**Comment by kamalkeshavani-aiinside** [comment]
This is kind of related to Tracking Superset Usage in Superset. We have created a virtual dataset using metadata tables to get such relations between database, dataset, charts & dashboards.

**Comment by mporracindie** [comment]
In case its useful, here are the query's i use now to do cleanups on my superset instance.

**Unused charts**
```sql
select s.*
from slices s
left join dashboard_slices ds on s.id = ds.slice_id
where ds.slice_id is null
```

**Unused datasets**
```sql
select t.*
from tables t
left join slices s on s.datasource_id = t.id
where s.datasource_id is null
```

**Comment by lzfxxx** [comment]
looks like related feature will be released at v2.1.0 https://github.com/apache/superset/pull/21760

**Reply by rusackas** [reply]
This works for the charts, but maybe we still need to keep the thread open for the sake of datasets? I'm not sure at the moment :P

**Comment by bsenthilbalu** [comment]
Is there any update on this feature to have an option to filter datasets which are not used by any charts?
